pkg/api/v1/defaults.go

```diff
@@ -117,6 +119,46 @@ func SetDefaults_VirtualMachine(obj *VirtualMachine) {
 	if obj.Spec.Domain.Features == nil {
 		obj.Spec.Domain.Features = &Features{}
 	}
+	// TODO: not fond of this code, suggestions welcome
+	if obj.Spec.Domain.Machine == nil {
+		obj.Spec.Domain.Machine = &Machine{}
```

Comment:
If we set a default always anyway, you can mache Machine a non-pointer and just check if Machine.Type is an empty string.

Reply:
done